ethereum/test/shutdownswitch.js

```diff
@@ -0,0 +1,471 @@
+        const initialized = new web3.eth.Contract(BridgeImplementationFullABI, TokenBridge.address);
+        // Don't clear the votes here.
+
+        // This assumes the previous test left us disabled.
+        assert.equal((await initialized.methods.numVotesToDisable().call()), 3)
+        assert.equal((await initialized.methods.enabledFlag().call()), false)
+
+        // Change one vote to enabled and our status should change back to enabled.
+        await initialized.methods.castShutdownVote(true).send({
+            value: 0,
+            from: accounts[0],
+            gasLimit: 2000000
+        });
+
+        assert.equal((await initialized.methods.numVotesToDisable().call()), 2)
+        assert.equal((await initialized.methods.enabledFlag().call()), true)
+
+        // And now the transfer should go through.
+        await initialized.methods.transferTokens(
+            TokenImplementation.address,
+            "10000",
+            "10",
+            "0x000000000000000000000000b7a2211e8165943192ad04f5dd21bedc29ff003e",
+            "0",
+            "0"
+        ).send({
+            value: 0,
+            from: accounts[0],
+            gasLimit: 2000000
+        });
+    })
+
+    it("should reject redeems when they are disabled", async function () {
+        // This test assumes the previous test initialized the vaa variable.
+        assert.ok(vaa != null)
+
+        const accounts = await web3.eth.getAccounts();
+        const initialized = new web3.eth.Contract(BridgeImplementationFullABI, TokenBridge.address);
+        await clearAllVotes(initialized, accounts);
+        assert.equal((await initialized.methods.enabledFlag().call()), true)
+
+        // Redeem of this transfer should not be blocked by shutdown switch. It will instead fail with
+        // "no quorum", but that's okay for this test, since that check happens after the enable check.
+        try {
+            await initialized.methods.completeTransfer("0x" + vaa).send({
+                value: 0,
+                from: accounts[0],
+                gasLimit: 2000000
+            })
+        } catch (error) {
+            assert.equal(error.message, "Returned error: VM Exception while processing transaction: revert no quorum")
+        }
+
+        // Cast three votes to disable transfers.
+        await initialized.methods.castShutdownVote(false).send({
+            value: 0,
+            from: accounts[0],
+            gasLimit: 2000000
+        });
+
+        assert.equal((await initialized.methods.numVotesToDisable().call()), 1)
+        assert.equal((await initialized.methods.enabledFlag().call()), true)
+
+        await initialized.methods.castShutdownVote(false).send({
+            value: 0,
+            from: accounts[1],
+            gasLimit: 2000000
+        });
+
+        assert.equal((await initialized.methods.numVotesToDisable().call()), 2)
+        assert.equal((await initialized.methods.enabledFlag().call()), true)
```

Comment:
I like how you are checking each step of the way here, on the state transitions.